src/solvers/NelderMead.jl

```diff
@@ -280,22 +284,45 @@ function step_solver!(mp::AbstractManoptProblem, s::NelderMeadState, ::Any)
         for i in 2:length(ind)
             retract!(
                 M,
-                s.population.pts[ind[i]],
-                s.population.pts[ind[1]],
+                s.population.pts[i],
+                s.population.pts[1],
                 inverse_retract(
-                    M,
-                    s.population.pts[ind[1]],
-                    s.population.pts[ind[i]],
-                    s.inverse_retraction_method,
+                    M, s.population.pts[1], s.population.pts[i], s.inverse_retraction_method
                 ),
                 s.σ,
                 s.retraction_method,
             )
             # update cost
-            s.costs[ind[i]] = get_cost(mp, s.population.pts[ind[i]])
+            s.costs[i] = get_cost(mp, s.population.pts[i])
         end
     end
     # store best
     s.p = s.population.pts[argmin(s.costs)]
     return s
 end
+
+mutable struct NelderMeadPopulationCriterion{TF<:Real,TX<:Real} <: StoppingCriterion
+    tol_f::TF
+    tol_x::TX
+    reason::String
+    function NelderMeadPopulationCriterion(tol_f::Real=1e-8, tol_x::Real=1e-8)
+        return new{typeof(tol_f),typeof(tol_x)}(tol_f, tol_x, "")
+    end
+end
+
+function (c::NelderMeadPopulationCriterion)(
+    mp::AbstractManoptProblem, s::NelderMeadState, i::Int
+)
+    (i == 0) && (c.reason = "") # reset on init
+    M = get_manifold(mp)
+    max_cdiff = maximum(cs -> abs(s.costs[1] - cs), s.costs[2:end])
+    max_xdiff = maximum(
+        p -> distance(M, s.population.pts[1], p, s.inverse_retraction_method),
+        s.population.pts[2:end],
+    )
+    if max_cdiff < c.tol_f && max_xdiff < c.tol_x
+        c.reason = "The simplex has shrunk below the assumed level (maximum cost difference is $max_cdiff, maximum point distance is $max_xdiff).\n"
```

Comment:
NIce. Could we maybe add the iteration number in this string? I find it helpful to read “After 57464 iterations” somewhere in a stopping criterion string.